### PR TITLE
Allow layers to be set for layout elevation profile item

### PR DESCRIPTION
**Pull request by nyalldawson**
Adds handling of visible layers and storing/retrieving these for layout elevation profile items

### CODE REVIEW

File: src/core/layout/qgslayoutitemelevationprofile.cpp [hunk 1]
```diff
@@ -401,6 +433,26 @@ bool QgsLayoutItemElevationProfile::readPropertiesFromElement( const QDomElement
...
+    const QDomElement layersElement = itemElem.firstChildElement( QStringLiteral( "layers" ) );
+    QDomElement layerElement = layersElement.firstChildElement( QStringLiteral( "layer" ) );
+    while ( !layerElement.isNull() )
+    {
+      const QString layerId = layerElement.attribute( QStringLiteral( "id" ) );
+      const QString layerName = layerElement.attribute( QStringLiteral( "name" ) );
+      const QString layerSource = layerElement.attribute( QStringLiteral( "source" ) );
+      const QString layerProvider = layerElement.attribute( QStringLiteral( "provider" ) );
+
+      QgsMapLayerRef ref = QgsMapLayerRef( layerId, layerName, layerSource, layerProvider );
```
Comment (nirvn) [inline src/core/layout/qgslayoutitemelevationprofile.cpp:448]: @nyalldawson , we don't have a util function to save / restore a QgsMapLayerRef to / from XML?

File: src/gui/layout/qgslayoutelevationprofilewidget.h [hunk 2]
```diff
@@ -59,6 +63,10 @@ class GUI_EXPORT QgsLayoutElevationProfileWidget: public QgsLayoutItemBaseWidget
     QPointer< QgsLayoutItemElevationProfile > mProfile = nullptr;
 
     QgsLayoutItemPropertiesWidget *mItemPropertiesWidget = nullptr;
+
+    std::unique_ptr< QgsLayerTree > mLayerTree;
+    QgsLayerTreeRegistryBridge *mLayerTreeBridge = nullptr;
+    QgsElevationProfileLayerTreeView *mLayerTreeView = nullptr;
```
Comment (nirvn) [inline src/gui/layout/qgslayoutelevationprofilewidget.h:69]: @nyalldawson , the object here is deleted when mLayerTree is deleted?

**Review by nirvn — APPROVED** [review]
Two Qs, all good.

File: src/core/layout/qgslayoutitemelevationprofile.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/layout/qgslayoutitemelevationprofile.cpp]: Good point, I've moved this to the QgsMapLayerRef code

Comment (nirvn) [reply src/core/layout/qgslayoutitemelevationprofile.cpp]: @nyalldawson , thanks.

**Pull request merged**